src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -372,6 +426,15 @@ private ReadEndsForMarkDuplicates buildReadEnds(final SAMFileHeader header, fina
             }
         }
 
+        if (useBarcodes) {
+            ((ReadEndsForMarkDuplicatesWithBarcodes) ends).barcode = getBarcodeValue(rec); // needed for frag sorting
```

Comment:
Would it be cleaner/clearer to cast to a variable and use it 3 times?

ReadEndsForMarkDuplicatesWithBarcodes endsWithBarcode = (ReadEndsForMarkDuplicatesWithBarcodes) ends;
endsWithBarcode.barcode = getBarcodeValue(rec); 
 if (rec.getFirstOfPairFlag()) {
    endsWithBarcode.readOneBarcode = getReadOneBarcodeValue(rec);
 } else {
    endsWithBarcode.readTwoBarcode = getReadTwoBarcodeValue(rec);
 }